rectools/models/nn/transformers/data_preparator.py

```diff
@@ -381,6 +400,17 @@ def transform_dataset_u2i(self, dataset: Dataset, users: ExternalIds) -> Dataset
         # Prepare new user id mapping
         rec_user_id_map = IdMap.from_values(interactions[Columns.User])
 
+        if context is not None:
+            if not (pd.Series(users).isin(context[Columns.User])).all():
+                raise ValueError("No context for all target users")
+            if context.duplicated(subset=Columns.User).any():
+                warnings.warn("Only the earliest row per user is used as context", UserWarning)
+                context = context_prep.get_context(context)
+            context[Columns.Item] = PADDING_VALUE  # External index pad element
+            context = context[context[Columns.User].isin(interactions[Columns.User])]
+            interactions = pd.concat([interactions, context])
```

Comment:
Probably I missed some part of code, but it seems that we haven't checked context columns, so they may be not the same as in the interactions